src/Sarif.Viewer.VisualStudio.Core/SarifViewerPackage.cs

```diff
@@ -203,12 +215,60 @@ private void SolutionEvents_OnBeforeCloseSolution(object sender, EventArgs e)
         {
             // stop watcher when the solution is closed.
             this.sarifFolderMonitor?.StopWatch();
+
+            var fileSystem = new FileSystem();
+
+            try
+            {
+                fileSystem.FileDelete(Path.Combine(GetDotSarifDirectoryPath(), "scan-results.sarif"));
+            }
+            catch (Exception) { }
         }
 
         private void SolutionEvents_OnAfterBackgroundSolutionLoadComplete(object sender, EventArgs e)
         {
             // start to watch when the solution is loaded.
             this.sarifFolderMonitor?.StartWatch();
+
+            this.JoinableTaskFactory.Run(async () => await RequestAnalysisResultsAsync());
+        }
+
+        private async Task RequestAnalysisResultsAsync()
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            // Currently this service only supports one result source.
+            if (this.resultSourceService == null)
+            {
+                var resultSourceFactory = new ResultSourceFactory(GetSolutionDirectoryPath(), this);
+                Result<IResultSourceService, ErrorType> result = await resultSourceFactory.GetResultSourceServiceAsync();
+
+                if (result.IsSuccess)
+                {
+                    this.resultSourceService = result.Value;
+                    this.resultSourceService.ResultsUpdated += this.ResultSourceService_ResultsUpdated;
+                }
+            }
+
+            await this.resultSourceService.RequestAnalysisScanResultsAsync();
+        }
+
+        private void ResultSourceService_ResultsUpdated(object sender, ResultsUpdatedEventArgs e)
+        {
+            string path = Path.Combine(GetDotSarifDirectoryPath(), e.LogFileName);
+            e.SarifLog.Save(path);
+        }
+
+        private static string GetSolutionDirectoryPath()
```

Comment:
>GetSolutionDirectoryPath [](http://example.com/codeflow?start=30&length=24)

nit there are couple functions defined in different projects, we should put it in a common place

Reply:
The other one I'm aware of is in SarifFolderMonitor and I didn't want to take a dependency on that. I'll file a bug to find a better central home for this.

Reply:
[https://dev.azure.com/mseng/1ES/_workitems/edit/1967398](https://dev.azure.com/mseng/1ES/_workitems/edit/1967398)